src/components/TagInput/TagInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { FormEvent, KeyboardEvent, useState } from 'react';
+
+import { TextInput } from '../Input';
+import { Tag, TagGroup } from '../Tag';
+import { Merge } from '../../types/polymophic';
+
+export type TagInputBaseProps = {
+  hint?: string;
+  label: string;
+  onTagsChange(tags: Array<string>): any;
+  tags?: Array<string>;
+};
+
+export type TagInputProps = Merge<React.InputHTMLAttributes<HTMLInputElement>, TagInputBaseProps>;
+
+export const TagInput = ({
+  hint,
+  label,
+  onTagsChange,
+  tags,
+  ...props
+}: TagInputProps) => {
+  const [input, setInput] = useState('');
+  const [values, setValues] = useState(tags);
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.key === "Enter" && input.trim()) {
+      e.preventDefault();
+      if (values?.includes(input.trim())) return;
+      const newValues = [...(values ?? []), input.trim()];
+      setValues(newValues);
+      setInput('');
+      onTagsChange(newValues);
+    }
+  };
+
+  const handleDeleteClick = (tag: string) => {
+    const newValues = [...(values ?? []).filter((el) => el !== tag)];
+    setValues(newValues);
+    onTagsChange(newValues);
+  };
+
+  return (
+    <div>
+      <TextInput
```

Comment:
Not sure it is a good idea to use a TextInput here espacially if features of it like message, messagetype are not used.
At least disableAutoValidation should be set to true